src/test/java/com/hubspot/jinjava/lib/filter/AbstractFilterTest.java

```diff
@@ -0,0 +1,163 @@
+package com.hubspot.jinjava.lib.filter;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import com.google.common.collect.ImmutableMap;
+import com.hubspot.jinjava.BaseInterpretingTest;
+import com.hubspot.jinjava.doc.annotations.JinjavaDoc;
+import com.hubspot.jinjava.doc.annotations.JinjavaParam;
+import com.hubspot.jinjava.interpret.JinjavaInterpreter;
+import java.math.BigDecimal;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.Test;
+
+public class AbstractFilterTest extends BaseInterpretingTest {
+  private ArgCapturingFilter filter;
+
+  public static class NoJinjavaDocFilter extends ArgCapturingFilter {}
+
+  @Test
+  public void itErrorsWhenNoJinjavaDoc() {
+    assertThatThrownBy(() -> new NoJinjavaDocFilter())
+      .hasMessageContaining("@JinjavaDoc must be configured");
+  }
+
+  @JinjavaDoc
+  public static class NoJinjavaParamsFilter extends ArgCapturingFilter {}
+
+  @Test
+  public void itDoesNotRequireParams() {
+    filter = new NoJinjavaParamsFilter();
+  }
+
+  @JinjavaDoc(
+    params = {
+      @JinjavaParam(value = "1st", desc = "1st"),
+      @JinjavaParam(value = "2nd", desc = "2nd"),
+      @JinjavaParam(value = "3rd", desc = "3rd")
+    }
+  )
+  public static class TwoParamTypesFilter extends ArgCapturingFilter {}
+
+  @Test
+  public void itSupportsMixingOfPositionalAndNamedArgs() {
+    filter = new TwoParamTypesFilter();
+
+    filter.filter(null, interpreter, new Object[] { "1" }, ImmutableMap.of("3rd", "3"));
+
+    assertThat(filter.parsedArgs).isEqualTo(ImmutableMap.of("1st", "1", "3rd", "3"));
+  }
+
+  @JinjavaDoc(
+    params = {
+      @JinjavaParam(
+        value = "boolean",
+        type = "boolean",
+        desc = "boolean",
+        required = true
+      ),
+      @JinjavaParam(value = "int", type = "int", desc = "int"),
+      @JinjavaParam(value = "long", type = "long", desc = "long"),
+      @JinjavaParam(value = "float", type = "float", desc = "float"),
+      @JinjavaParam(value = "double", type = "double", desc = "double"),
+      @JinjavaParam(value = "number", type = "number", desc = "number"),
+      @JinjavaParam(value = "object", type = "object", desc = "object"),
+      @JinjavaParam(value = "dict", type = "dict", desc = "dict")
+    }
+  )
+  public static class AllParamTypesFilter extends ArgCapturingFilter {}
+
+  @Test
+  public void itParsesNumericAndBooleanInput() {
+    filter = new AllParamTypesFilter();
+
+    Map<String, Object> kwArgs = new HashMap<>();
```

Comment:
You could use `ImmutableMap` here for a more fluent interface and less repetition.

Reply:
Unfortunately `ImmutableMap.of` has a limit of 5 keys so forced to use this

Reply:
You could use `ImmutableMap.builder()`.

Reply:
Done